src/components/DataCard/DataCard.tsx

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { useMemo } from "react";
+import { Icon } from "@/components/Icon/Icon";
+import { macroThemes } from "@/utils/constants";
+import { IMetadata, MacroTheme } from "@/utils/interfaces";
+
+const normalizeKey = (value?: string) =>
+  value
+    ?.normalize("NFD")
+    .replace(/[\u0300-\u036f]/g, "")
+    .replace(/[\s-]+/g, "_")
+    .toLowerCase() ?? "";
+
+export const DataCard = ({
+  post,
+  themes,
+}: {
+  post: IMetadata & { tagSlugs?: string[] };
+  themes: MacroTheme[];
+}) => {
+  const primaryFile = post.files?.[0];
+  const additionalFiles = post.files?.slice(1) ?? [];
+
+  const themeLookup = useMemo(() => {
```

Comment:
Not really sure if this needs to be a hook, since we're just going to run this once this memo ends up being not useful, but forces us to use "use client", maybe would be better to remove it and make it a regular function an then call it